code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubUtils.java

```diff
@@ -0,0 +1,150 @@
+package io.cattle.platform.iaas.api.auth.github;
+
+import io.cattle.platform.token.TokenException;
+import io.cattle.platform.token.TokenService;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class GithubUtils {
+
+    private TokenService tokenService;
+
+    public String validateAndFetchGithubToken(String jwt) {
+        if(StringUtils.isEmpty(jwt)) {
+            return null;
+        }
+        String toParse = null;
+        String[] tokenArr = jwt.split("\\s+");
+        if (tokenArr.length == 2) {
+            if (!StringUtils.equalsIgnoreCase("bearer", StringUtils.trim(tokenArr[0]))) {
+                return null;
+            }
+            toParse = tokenArr[1];
+        } else if (tokenArr.length == 1) {
+            toParse = tokenArr[0];
+        } else {
+            return null;
+        }
+        Map<String, Object> jsonData;
+        try {
+            jsonData = tokenService.getJsonPayload(toParse, true);
+        } catch (TokenException e) { // in case of invalid token
+            return null;
+        }
+        return (String) jsonData.get("access_token");
```

Comment:
Casting is bad.  There are safer ways to do this, like ObjectUtils.toString(val, null).  Don't forget the null otherwise you get a string like "null".